app/forms/keywords_form.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+require 'csv'
+
+class KeywordsForm
+  include ActiveModel::Model
+
+  validates_with KeywordsFormValidator
+
+  attr_reader :file, :user
+
+  def initialize(user)
+    @user = user
+  end
+
+  def save(csv_file)
+    @file = csv_file
```

Comment:
Yeah, I agree with @carryall.
Because now our keywords are clean now, I think it's better to save in the controller.
So now we can remove the service about saving to database here 👍

Reply:
I like it too, I went ahead and changed it up 👷🏾 Updated in https://github.com/aetheryna/google-scraper-ruby/pull/34/commits/439908c2734f5d8ffecd8a64310ecddee29f9cf6